node/cn/tracers/internal/tracers/revert_tracer.js

```diff
@@ -34,7 +35,16 @@
     },
 
     // step is invoked for every opcode that the VM executes.
-    step: function(log, db) { },
+    step: function(log, db) {
+        if(this.callerAddress == "")
+            this.callerAddress = "0x" + log.contract.getAddress();
+        if(log.op === undefined)
+            return;
+        var op = log.op.toString();
+        if(op == "CALL" || op == "CALLCODE" || op == "DELEGATECALL" || op == "STATICCALL" ) {
+            this.callerAddress = "0x" + log.stack.peek(1).toString(16);
+        }
+    },
```

Comment:
This code saves the last callee address.
Therefore, if revert error is happened after the address contract was returned, this code shows the wrong contract address.

Reply:
good point! Thanks. I will check that. :)